nipype/interfaces/io.py

```diff
@@ -449,20 +448,9 @@ def _check_s3_base_dir(self):
             s3_flag = False
             return s3_flag, bucket_name
 
-        # Explicitly lower-case the "s3"
-        if base_directory.lower().startswith(s3_str):
-            base_dir_sp = base_directory.split("/")
-            base_dir_sp[0] = base_dir_sp[0].lower()
-            base_directory = "/".join(base_dir_sp)
-
-        # Check if 's3://' in base dir
-        if base_directory.startswith(s3_str):
-            # Expects bucket name to be 's3://bucket_name/base_dir/..'
-            bucket_name = base_directory.split(s3_str)[1].split("/")[0]
-            s3_flag = True
-        # Otherwise it's just a normal datasink
-        else:
-            s3_flag = False
+        s3_flag = base_directory.lower().startswith(s3_str)
+        if s3_flag:
+            bucket_name = base_directory[len(s3_str):].split("/")[0]
```

Comment:
No real need to find more than the first `/`.

```suggestion
            bucket_name = base_directory[len(s3_str):].split("/", 1)[0]
```

Reply:
apparently `partition` is a faster alternative that only splits once